tests/ssl-opt.sh

```diff
@@ -1580,7 +1581,7 @@ run_test    "Context serialization, re-init, both serialize, with CID" \
 requires_config_enabled MBEDTLS_SSL_DTLS_CONNECTION_ID
 run_test    "Connection ID: Cli enabled, Srv disabled" \
             "$P_SRV debug_level=3 dtls=1 cid=0" \
-            "$P_CLI debug_level=3 dtls=1 cid=1 cid_val=deadbeef" \
+            "$P_CLI debug_level=3 dtls=1 cid=1 cid_val=dead" \
```

Comment:
Question: is there any particular reason why this test and the next one were modified to use a smaller CID, while the others use `requires_config_disabled`?

Reply:
The `MBEDTLS_SSL_CONF_CID_LEN` restricts the usage to only that hardcoded length, if we want the tests to run with baremetal, it should be same length. The tests that are having the `requires_config_disabled` is for tests that are using connection id of length 0. Those tests cannot be altered because changing 0-length connection id would ruin their cause, so they need to be skipped.

Reply:
OK, thanks for explaining, it's all clear now!